Kudu.Core/SourceControl/Git/LibGit2SharpRepository.cs

```diff
@@ -0,0 +1,396 @@
+        {
+            using (var repo = new LibGit2Sharp.Repository(RepositoryPath))
+            {
+                repo.Checkout(id, new CheckoutOptions { CheckoutModifiers = CheckoutModifiers.Force });
+            }
+        }
+
+        public void Update()
+        {
+            Update("master");
+        }
+
+        public void Push()
+        {
+            using (var repo = new LibGit2Sharp.Repository(RepositoryPath))
+            {
+                var masterBranch = repo.Branches["master"];
+                repo.Network.Push(masterBranch);
+            }
+        }
+
+        public void FetchWithoutConflict(string remoteUrl, string branchName)
+        {
+            ITracer tracer = _tracerFactory.GetTracer();
+            try
+            {
+                using (var repo = new LibGit2Sharp.Repository(RepositoryPath))
+                {
+                    var trackedBranchName = string.Format("{0}/{1}", _remoteAlias, branchName);
+                    var refSpec = string.Format("+refs/heads/{0}:refs/remotes/{1}", branchName, trackedBranchName);
+
+                    // Configure the remote
+                    var remote = repo.Network.Remotes.Add(_remoteAlias, remoteUrl, refSpec);
+
+                    // This will only retrieve the "master"
+                    repo.Network.Fetch(remote);
+
+                    // Optionally set up the branch tracking configuration
+                    var trackedBranch = repo.Branches[trackedBranchName];
+                    if (trackedBranch == null)
+                    {
+                        throw new BranchNotFoundException(branchName, null);
+                    }
+                    var branch = repo.Branches[branchName] ?? repo.CreateBranch(branchName, trackedBranch.Tip);
+                    repo.Branches.Update(branch,
+                        b => b.TrackedBranch = trackedBranch.CanonicalName);
+
+                    //Update the raw ref to point the head of branchName to the latest fetched branch
+                    UpdateRawRef(string.Format("refs/heads/{0}", branchName), trackedBranchName);
+
+                    // Now checkout out our branch, which points to the right place
+                    Update(branchName);
+                }
+            }
+            finally
+            {
+                using (var repo = new LibGit2Sharp.Repository(RepositoryPath))
+                {
+                    repo.Network.Remotes.Remove(_remoteAlias);
+                }
+            }
+        }
+
+        public void Clean()
+        {
+            using (var repo = new LibGit2Sharp.Repository(RepositoryPath))
+            {
+                repo.RemoveUntrackedFiles();
+            }
+        }
+
+        public void CreateOrResetBranch(string branchName, string startPoint)
+        {
+            using (var repo = new LibGit2Sharp.Repository(RepositoryPath))
+            {
+                var branch = repo.Branches[branchName];
+                if (branch == null)
```

Comment:
Another logic flow for this api could be ...

1. if branch not exists, create branch.
2. checkout branch.
3. Reset hard with commit.

Less if else this way.  Thought?